crates/loader/src/local/mod.rs

```diff
@@ -71,6 +85,10 @@ async fn prepare(
 ) -> Result<Application<CoreComponent>> {
     let info = info(raw.info, &src);
 
+    if let Err(e) = error_on_duplicate_ids(raw.components.clone()) {
```

Comment:
I think you can just `error_on_duplicate_ids(raw.components.clone())?` here?

Reply:
Ah, ignore https://github.com/fermyon/spin/pull/370/files#r857180717

Adding the `?` makes my issues disappear 🙂 I _don't think_ I really understand why yet - but happy to push these changes up. Thanks!

Reply:
Basically:
> The `?` placed after a `Result` value is defined to work in almost the same way as a `match` expressions defined to handle the `Result` values.

Here are a few resources on Rust errors:
- https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator
- https://blog.burntsushi.net/rust-error-handling/

Reply:
That's super helpful, thanks! And thanks for guiding me through this PR. It's quite trivial, but I learned a lot!